Problem Statement 1:
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of
15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.

In [2]:
import scipy.stats as st
import numpy as np

In [3]:
def z_stat(samp_mean,pop_mean,se):
    return (samp_mean-pop_mean)/se
def t_stat(samp_mean,pop_mean,se):
    return (samp_mean-pop_mean)/se
def sample_error(std,n):
    return std/(np.sqrt(n))
def z_test_pop_proportions(p1,p2,se):
    return ((p1-p2)/se)
def sample_error_pop_proportions(p1,p2,n1,n2):
    return np.sqrt(((p1*(1-p1))/n1)+((p2*(1-p2))/n2))


In [4]:

alpha=0.05
#Since the sample size >30,we use z test
z=z_stat(samp_mean=108,pop_mean=100,se=sample_error(std=15,n=36))
p=2*(1-st.norm.cdf(np.absolute(z)))
if p<alpha:
    print("Null Hypothesis is rejected(The raw corn starch had an effect)")
else:
    print("Null Hypothesis is accepted(The raw cornstarch has no effect)")

Null Hypothesis is rejected(The raw corn starch had an effect)


Problem Statement 2:
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?

In [5]:
def z_test_pop_proportions(p1,p2,se):
    return ((p1-p2)/se)
def sample_error_pop_proportions(p1,p2,n1,n2):
    return np.sqrt(((p1*(1-p1))/n1)+((p2*(1-p2))/n2))

In [ ]:
alpha=0.05

In [6]:
sample_error_pop_proportions= sample_error_pop_proportions(0.48,0.53,100,100)

In [7]:
z_test_pop_proportions=z_test_pop_proportions(0.48,0.53,sample_error_pop_proportions)

In [9]:
p=st.norm.cdf(z)
if p<alpha:
    print("Null Hypothesis is rejected")
else:
    print("Null Hypothesis is accepted")

Null Hypothesis is accepted


Problem Statement 3:
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?

In [15]:
z=(1100-1026)/209

In [16]:
z

0.35406698564593303

This means that your score was .354 std devs above the mean.